pkg/didcomm/protocol/issuecredential/service.go

```diff
@@ -63,14 +67,40 @@ type metaData struct {
 	inbound  bool
 	// keeps offer credential payload,
 	// allows filling the message by providing an option function
-	offerCredential   OfferCredential
-	proposeCredential ProposeCredential
+	offerCredential   *OfferCredential
+	proposeCredential *ProposeCredential
 	// err is used to determine whether callback was stopped
 	// e.g the user received an action event and executes Stop(err) function
 	// in that case `err` is equal to `err` which was passing to Stop function
 	err error
 }
 
+// Action contains helpful information about action
+type Action struct {
+	// protocol state machine identifier
+	PIID string
+	Msg  service.DIDCommMsgMap
+}
+
+// Opt describes option signature for the Continue function
+type Opt func(md *metaData)
+
+// WithProposeCredential allows to provide ProposeCredential message
```

Comment:
(optional/low priority)

"allows to provide" sounds odd to me... consider "allows providing" (similar to how you used "allows proceeding" in another area in AllowContinue method) or "allows for the ability to provide a ProposeCredential messsage"? (also for the other "allows to provide" messages in this PR)